### DISCUSSION TITLE
Should I be surprised to see Magenta and its fellows listed as subclasses of color rather than reference data instances?

### DISCUSSION THREAD

**Discussion by ericatail** [CommonCoreOntology/CommonCoreOntologies#837] [Ontology Terms and Class Schema]
Should I be surprised to see Magenta and its fellows listed as subclasses of Color rather than reference data instances?

Thanks!

**Comment by swartik** [comment]
Well, two distinct independent continuants would not bear the same magenta color individual. Each would bear a separate color so you could associate both the independent continuant and the color with a process individual. You would then use that process individual to record when it was observed that the individual was magenta, who made the observation, etc.

There are other considerations. You might have each Color individual concretize a measurement information content entity (MICE) to describe the wavelength range of the color. In this formulation, a green color has a wavelength between 495-570 nm (according to Wikipedia). You could use that to make assertions about the MICE a Color subclass concretizes.

But as in all uses of ontology, the real answer to your question is: what do you want to model?

**Comment by ericatail** [comment]
Thanks for your reply.

If colors were pure single-frequency EM radiation, then we might give them EM frequency intervals and be done with it.
But many are, of course, superimposed waveforms.

But a EM sensor could give us a pure green frequency in some cases.  And a verbal narative that we perform entity extraction on would parse the word "green" and might instantiate a shared _Green_ instance with your standard green interval.

But my customer isn't giving me any color sensor data,  So to answer your question, the latter case is what I want to model.

I apologize, I can't stop thinking of that interval green case or the simple Magenta color token as an instance.   _Green_ would be a shared instance with an attached freq. interval. that everyone would use - I call that shared reference data - like the _name_ shared instance _John_.
So would independent continuants actually need to bear their own color individuals?  I didn't follow why processes come into the discussion.   I'm not sure why the process couldn't just reference a shared individual color.  But I'm rather new to CCO.

Thanks!
-Eric

**Comment by ericatail** [comment]
To my mind, the very fact that we are discussing frequency ranges, superimposed wave forms and such means that we need an instance to hang such data off of.

And it seems inefficient  to have multiple instances of your _cco:Green_ when one instance of _cco:Color_ will do.  Sorry if I'm missing something obvious.

**Comment by ericatail** [comment]
ref:Green a cco:Color .

**Comment by cameronmore** [comment]
There are different ways of understanding this. Unlike measurement units, which are shared reference data (we all point to the same 'Meter' when measuring), green and other colors are instances of dispositions, and so each instance of color you encounter is going to be a unique, different instance of a disposition. Just as two people who weigh 150lbs do not weigh **the same** 150lbs, they are different instances of weight but the measurement of their weight may have the same literal value.

Another way to think about green as a disposition; green is a color, and a color is a phenomenon consisting of hue, saturation, and brightness of how light behaves. Green is **just** a subset of all those instances of colors which have a particular hue/brightness/saturation that namely arises 'in the visible spectrum, typically between 520 to 560 nanometers.' Instances of colors are always instances of colors on this visual spectrum, and these subclasses are approximate attempts to group those instances in terms of our ability to differentiate them.

Or, to put it another way, it is true that the green in one tree is not the same in another tree, but what is the same is that they are both particular, different instances of cco:Colors with a certain similarity based on their hue/brightness/saturation, being in the same portion of the light spectrum.

I'm happy to work through this issue, please let me know if I am not capturing your concern, thank you, Cameron

**Comment by ericatail** [comment]
Hey Cameron!

Thanks for your reply!  It's good to bump into you over here.

Maybe I'm too focused on the data I have to work with.   I suppose that green could be thought of as a collection of the individual green wavelengths you speak of and I've claimed before that any collection could be declared to be a class (or subclass).

Right now, all my customer needs is the enumerated type green.  But since we all want our our modeling to endure changes in our level of sophistication, I modeled it as an instance.    When we start to care about wavelengths,, hue, saturation, etc., hanging this data off the subclass _Green_ seems rather odd to my software engineering sensibilities.

But I have bigger fish to fry, I I probably just need to get used to using the subclass _Green_ .  It will work as a sort of enumerated type and I would guess that will cover my modest data set needs for the next decade or so.



Thanks again!

-Eric
